### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by MGaetan89**
This PR moves `sickbeard.client.http_error_code`, `sickbeard.helpers.tryInt` and `sickbeard.helpers. codeDescription` to the `sickrage` module.

`sickbeard.client.http_error_code` had a TODO about duplicate key handling. I fixed that by setting a list to the HTTP status code that can have multiple values.

The log line in `http_code_description` (previously named `codeDescription`) is commented because there is a cyclic dependency. This can be restored once the logger has been rewritten.

### CODE REVIEW

File: sickrage/helper/common.py [hunk 1]
```diff
@@ -21,6 +21,84 @@
...
+    409: 'Conflict',
+    410: 'Gone',
+    411: 'Length Required',
+    412: 'Precondition Failed',
+    413: 'Request Entity Too Large',
+    414: 'Request-URI Too Long',
+    415: 'Unsupported Media Type',
+    416: 'Requested Range Not Satisfiable',
+    417: 'Expectation Failed',
+    418: 'Im a teapot',
```
Comment (duramato) [inline sickrage/helper/common.py:53]: Do webservers even return this error code? lol

Comment (MGaetan89) [inline sickrage/helper/common.py:53]: I don't think so :p

Comment (Arcanemagus) [inline sickrage/helper/common.py:53]: There are implementations: https://en.wikipedia.org/wiki/Hyper_Text_Coffee_Pot_Control_Protocol :stuck_out_tongue:

**Comment by miigotu** [comment]
@MGaetan89 need some tests for ex, some ex calls are returning None (I think when it is just a string with no args?) such as when github api limit is hit or username/password are wrong, and some errors in postProcessor (actually processTV I think)

If I change the ex call to a str call they show just fine.

**Comment by MGaetan89** [comment]
OK, I will see if I can write proper tests for that in my next PR.

File: sickrage/helper/common.py (hunk 1 above)
Comment (MGaetan89) [inline sickrage/helper/common.py:53]: O.o

**Comment by labrys** [comment]
@MGaetan89 When you add to the test, you might also want to add unicode test cases, such as unicode in the response, etc.

File: sickrage/helper/common.py (hunk 1 above)
Comment (duramato) [inline sickrage/helper/common.py:53]: lol @ http://error418.net/

Comment (Arcanemagus) [inline sickrage/helper/common.py:53]: Wow, that page actually responds with the number, but not the text:
image

**Comment by MGaetan89** [comment]
@labrys I always add Unicode test cases

**Comment by labrys** [comment]
I've noticed, I just didn't see any in the http responses, but I was tired, might have missed them. I appreciate that someone else is giving our testing some TLC. :+1:

**Comment by labrys** [comment]
@MGaetan89 I'd like to export test cases out of the tests and put them in a file (likely json) so they can be easily updated and added to as well as reused between tests.

**Comment by MGaetan89** [comment]
I thought about that, but as it's my first tests experience in Python, I just followed the structure that was already in place.
If you want/know to change that, I support you :)

**Comment by labrys** [comment]
I've been emulating whats there myself, but its been bugging (pun intended) me :)
